### ISSUE TITLE
[BUG] Failed to build from source because of missing #include directive

### ISSUE THREAD

**Issue by tonyxty** [tensorflow/tensorflow#19575]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.8.0, tf.GIT_VERSION is 'unknown' (I compiled from commit dcb10b1d557168646204239bea6ca5bf1abc40a3)
- **Python version**: Python 3.6.5
- **Bazel version (if compiling from source)**: 0.13.1
- **GCC/Compiler version (if compiling from source)**: 7.3.0
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package

### Describe the problem
When compiling tensorflow from source following the instructions here (https://www.tensorflow.org/install/install_sources#build_the_pip_package), I encountered the following C++ error:
```
ERROR: /path/to/tensorflow/tensorflow/python/BUILD:310:1: C++ compilation of rule '//tensorflow/python:cpp_python_util' failed (Exit 1)
tensorflow/python/util/util.cc:276:16: error: 'function' in namespace 'std' does not name a template type
     const std::function<int(PyObject*)>& is_sequence_helper,
                ^~~~~~~~
tensorflow/python/util/util.cc:276:24: error: expected ',' or '...' before '<' token
     const std::function<int(PyObject*)>& is_sequence_helper,
                        ^
tensorflow/python/util/util.cc: In function 'bool tensorflow::swig::{anonymous}::FlattenHelper(PyObject*, PyObject*, int)':
tensorflow/python/util/util.cc:280:16: error: 'is_sequence_helper' was not declared in this scope
   int is_seq = is_sequence_helper(nested);
                ^~~~~~~~~~~~~~~~~~
tensorflow/python/util/util.cc:280:16: note: suggested alternative: 'IsSequenceHelper'
   int is_seq = is_sequence_helper(nested);
                ^~~~~~~~~~~~~~~~~~
                IsSequenceHelper
[... other C++ mumbling ...]
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
```

### Source code / logs
Adding ``#include <functional>`` to the file solves the issue.

**Comment by dburkhardt** [comment]
I also had this issue compiling on an Arch machine with similar software versions as above.

**Comment by tonyxty** [comment]
This has been fixed by commit 2561c4000afcead84823ebead70498533e5ebebb.  Closing.
